support/scripts/uk-libinfo.py

```diff
@@ -0,0 +1,272 @@
+			for rec in self._recs:
+				str_type = "{:}:".format(rec.type_description())
+				ret += "{:30}{:}\n".format(str_type, rec.str_data())
+		else:
+			ret = "{:}:\n".format(self._libname)
+			for rec in self._recs:
+				str_type = "{:}:".format(rec.type_description())
+				ret += "  {:28}{:}\n".format(str_type, rec.str_data())
+		return ret
+
+	def get_recs(self):
+		return self._recs
+
+	def libname(self):
+		return self._libname
+
+def parse_uklibinfo(data, byteorder=sys.byteorder):
+	if not data:
+		raise Exception("No uklibinfo data found for parsing")
+	hdr_hdr_len = (4 + 2) # sizeof(__u32) + sizeof(__u16)
+	rec_hdr_len = (2 + 4) # sizeof(__u16) + sizeof(__u32)
+	seek = 0
+	left = len(data)
+	found = []
+
+	# We seek and remember the number of left bytes so that
+	# we avoid copying sub data
+	while left >= hdr_hdr_len:
+		# struct uk_libid_info_hdr
+		hdr_len     = int.from_bytes(data[seek:seek + 4],
+					     byteorder)
+		seek       += 4 # sizeof(__u32 len)
+		if (hdr_len > left):
+			raise Exception("Invalid header size at byte position {}".format(seek))
+		hdr_version = int.from_bytes(data[seek:seek + 2],
+					     byteorder)
+		seek       += 2 # sizeof(__u16 version)
+		left       -= hdr_len
+		rec_seek    = seek
+		rec_left    = hdr_len - hdr_hdr_len
+		seek       += rec_left
+
+		if not hdr_version == 1:
+			sys.stderr.write("Skipping unsupported library information version {}\n".format(hdr_version))
+			continue
+
+		hdr = UKLibraryInfoHeader(hdr_version, byteorder)
+		found.append(hdr)
+
+		# struct uk_libid_info_rec
+		while rec_left >= rec_hdr_len:
+			rec_type   = int.from_bytes(data[rec_seek:rec_seek + 2],
+						    byteorder)
+			rec_seek  += 2 # sizeof(__u16 type)
+			rec_len    = int.from_bytes(data[rec_seek:rec_seek + 4],
+						    byteorder)
+			rec_seek  += 4 # sizeof(__u32 len)
+			if (rec_len > rec_left):
+				raise Exception("Invalid record size at byte position {}".format(rec_seek))
+			rec_data   = data[rec_seek:
+					  rec_seek + (rec_len - rec_hdr_len)]
+			rec_seek  += len(rec_data)
+			rec_left  -= rec_len
+
+			hdr.add_rec(UKLibraryInfoRecord(rec_type, rec_data,
+							byteorder))
+	return found
+
+# Load uklibinfo data from ELF section
+def uklibinfo_elf_load(path):
+	elf_sec_name = ".uk_libinfo"
+	fd           = open(path, 'rb')
+	elffile      = ELFFile(fd)
+	section      = None
+	for s in elffile.iter_sections():
+		if s.name == elf_sec_name:
+			section = s
+			break
+	if not section:
+		raise Exception("Section " + elf_sec_name + " not found")
+	uklibinfo = section.data()
+	fd.close()
+	return uklibinfo
+
+##
+## MAIN
+##
+if __name__ == '__main__':
+	parser = argparse.ArgumentParser(description='Unikraft library metadata tool')
+	parser.add_argument("file", help="Kernel or library to investigate")
+	(opt, rem_args) = parser.parse_known_args()
+
+	try:
+		uklibinfo = uklibinfo_elf_load(opt.file)
+	except Exception as e:
+		print("{}: Failed to load uklibinfo data: {}".format(opt.file, str(e)))
```

Comment:
This fails for me when looking at an arm64 binary: `build/unikraft_qemu-arm64: Failed to load uklibinfo data: Magic number does not match`

I think this is a limitation of the elf loader ([User guide](https://github.com/eliben/pyelftools/wiki/User's-guide)), not the rest of the implementation:
> The current focus of the library is on Intel's x86 and x64 architectures, with some ARM support added recently.